plugins/quantum_k_means/backend/cluster_algos/negative_rotation.py

```diff
@@ -0,0 +1,174 @@
+from .clustering import Clustering
+import pennylane as qml
+import numpy as np
+from typing import List
+
+
+class NegativeRotationQuantumKMeans(Clustering):
+
+    def __init__(
+        self,
+        backend: qml.Device,
+        tol,
+        max_runs
+    ):
+        super(NegativeRotationQuantumKMeans, self).__init__(backend, tol, max_runs)
+        # Number of qbits needed to calculate "distance" between one data point and one centroid
+        self.needed_qbits = 1
+
+    def init_centroids(self, k: int) -> np.ndarray:
+        """
+        Gets k random 2D points, then standardize and normalize them.
+        """
+        return self.Normalize(np.random.uniform(size=(k, 2), low=-1, high=1))
+
+    def prep_centroids(self, centroids) -> np.ndarray:
+        """
+        Normalizes the centroids.
+        """
+        return self.Normalize(centroids)
+
+    def calculate_angles(self, cartesian_points):
+        """
+        Calculates the angle between the 2D vetors and the base vector.
+        The cartesian points are given in a tuple format (x, y).
+        """
+        base_vector = np.array([1, 0])
+        angles = []
+        for d_point in cartesian_points:
+            if d_point[1] < 0:
+                angles.append(2 * np.pi - np.arccos(np.dot(d_point, base_vector)))
+            else:
+                angles.append(np.arccos(np.dot(d_point, base_vector)))
+        return np.array(angles)
+
+    def prep_data(self, data) -> np.ndarray:
+        """
+        Prepares data, by standardizing and normalizing it.
+        """
+        return self.Normalize(self.Standardize(data))
+
+    def prep_data_for_circuit(self, data: np.ndarray) -> np.ndarray:
+        """
+        Prepare data for the quantum circuit by calculating the necessary angles for the algorithm.
+        """
+        return self.calculate_angles(data)
+
+    def quantum_circuit(self, wires_to_use: List[int], data_angle: float, centroid_angle: float):
+        qml.RY(data_angle, wires=wires_to_use)  # data angle rotation
+        qml.RY(-centroid_angle, wires=wires_to_use)  # negative centeroid angle rotation
+
+    def execute_circuit(self, current_distances_calculated, data_angles: List[float], centroid_angles: np.ndarray):
+        @qml.qnode(self.backend)
+        def circuit():
+            wires_to_measure = []
```

Comment:
Only added a line after the function circuit(), because Black removes the empty line, if I put it infront of the function. I can put one back in though.